homeassistant/components/google_assistant/trait.py

```diff
@@ -2345,3 +2346,74 @@ def query_attributes(self):
                     {"name": data[0], "rawValue": self.state.state}
                 ]
             }
+
+
+@register_trait
+class AppSelectorTrait(_Trait):
+    """Trait to select apps.
+
+    https://developers.google.com/assistant/smarthome/traits/appselector
+    """
+
+    name = TRAIT_APPSELECTOR
+    commands = [COMMAND_APP_INSTALL, COMMAND_APP_SEARCH, COMMAND_APP_SELECT]
+
+    @staticmethod
+    def supported(domain, features, device_class, _):
+        """Test if state is supported."""
+        if domain == media_player.DOMAIN and (
+            features & media_player.SUPPORT_SELECT_SOURCE
+        ):
+            return True
+
+        return False
```

Comment:
Honestly, I think this needs more thought into the media player integration in general. Source is often misused for this.

Therefore I think we should not build on top of it.

Reply:
But we have already built on top of it? the InputSelector trait already does this for Apps(and not just inputs), again, this is adding more natural language support for Google Assistant, not adding something new.

Reply:
I kinda agree, however, I honestly think we should add the notion of app to these entities (or solve that in the future) in some way. And if we do that, there would be a distinction between those.

In case the above would be implemented in Google Assistant, it would cause a breaking change (or having to support both for the future as well). Which is not a wish-ed-for situation.

Reply:
If this was opt in(as in you put some more data in configuration.yaml) to enable it for specific inputs(or even just a global opt in), would that be acceptable?

Reply:
No, that would still hit the same concerns.

Reply:
I suppose this derails things a but. But... Many TV sort of see a source (being hdmi1,2,3..) as a channel of it's hardware input application. Soo.. 

Having a separate application list could make sense. Then we could sort of generalize source to mean channel of active application.

If it's the input application, it lists HDMI1,2,3..
If it's the tuner application, it lists channel 1, 2 3..
If it's YouTube, it might list subscribed channels.

But, it sort of brings us back to media browser, since you would not easily say switch to HDMI1. Since the list of sources would change based on active application.

Reply:
I was thinking about this more, and for the concept of breaking changes, whatever is done for media_player entities is going to be far bigger of an issue than voice commands that sound unnatural(that I would expect almost no one to use) not working in the distant future.

Your still gonna deal with the exact same breaking change regardless though, since every source is currently broadcast as an input to Google.
Whatever breaking change that applies for InputSelector would inversely apply for AppSelector...